e2e/acl/acl_token_test.go

```diff
@@ -0,0 +1,163 @@
+	// get an error.
+	tokenTTLLow := api.ACLToken{
+		Name:          "e2e-acl-" + uuid.Short(),
+		Type:          "client",
+		Policies:      []string{customNamespacePolicy.Name},
+		ExpirationTTL: minTokenExpiryDur / 2,
+	}
+	aclTokenCreateResp, _, err := nomadClient.ACLTokens().Create(&tokenTTLLow, nil)
+	require.ErrorContains(t, err, fmt.Sprintf(
+		"expiration time cannot be less than %s in the future", minTokenExpiryDur))
+	require.Nil(t, aclTokenCreateResp)
+
+	// Attempt to create a token with a higher than acceptable TTL and ensure
+	// we get an error.
+	tokenTTLHigh := api.ACLToken{
+		Name:          "e2e-acl-" + uuid.Short(),
+		Type:          "client",
+		Policies:      []string{customNamespacePolicy.Name},
+		ExpirationTTL: 8766 * time.Hour,
+	}
+	aclTokenCreateResp, _, err = nomadClient.ACLTokens().Create(&tokenTTLHigh, nil)
+	require.ErrorContains(t, err, fmt.Sprintf(
+		"expiration time cannot be more than %s in the future", maxTokenExpiryDur))
+	require.Nil(t, aclTokenCreateResp)
+
+	// Create an ACL token that has a fairly standard TTL that will allow us to
+	// make successful calls without it expiring.
+	tokenNormalExpiry := api.ACLToken{
+		Name:          "e2e-acl-" + uuid.Short(),
+		Type:          "client",
+		Policies:      []string{customNamespacePolicy.Name},
+		ExpirationTTL: 10 * time.Minute,
+	}
+	tokenNormalExpiryCreateResp, _, err := nomadClient.ACLTokens().Create(&tokenNormalExpiry, nil)
+	require.NoError(t, err)
+	require.NotNil(t, tokenNormalExpiryCreateResp)
+	require.Equal(t,
+		*tokenNormalExpiryCreateResp.ExpirationTime,
+		tokenNormalExpiryCreateResp.CreateTime.Add(tokenNormalExpiryCreateResp.ExpirationTTL))
+
+	cleanUpProcess.add(tokenNormalExpiryCreateResp.AccessorID, aclTokenTestResourceType)
+
+	// Add the token to our query options and ensure we can now list jobs with
+	// the default namespace.
+	defaultNSQueryMeta.AuthToken = tokenNormalExpiryCreateResp.SecretID
+
+	jobListResp, _, err := nomadClient.Jobs().List(&defaultNSQueryMeta)
+	require.NoError(t, err)
+	require.Empty(t, jobListResp)
+
+	// Create an ACL token with the lowest expiry TTL possible, so it will
+	// expire almost immediately.
+	tokenQuickExpiry := api.ACLToken{
+		Name:          "e2e-acl-" + uuid.Short(),
+		Type:          "client",
+		Policies:      []string{customNamespacePolicy.Name},
+		ExpirationTTL: minTokenExpiryDur,
+	}
+	tokenQuickExpiryCreateResp, _, err := nomadClient.ACLTokens().Create(&tokenQuickExpiry, nil)
+	require.NoError(t, err)
+	require.NotNil(t, tokenQuickExpiryCreateResp)
+	require.Equal(t,
+		*tokenQuickExpiryCreateResp.ExpirationTime,
+		tokenQuickExpiryCreateResp.CreateTime.Add(tokenQuickExpiryCreateResp.ExpirationTTL))
+
+	cleanUpProcess.add(tokenQuickExpiryCreateResp.AccessorID, aclTokenTestResourceType)
+
+	// Block the test (sorry) until the token has expired.
+	timer, timerStopFn := helper.NewSafeTimer(tokenQuickExpiry.ExpirationTTL)
+	defer timerStopFn()
+
+	select {
+	case <-timer.C:
+	}
```

Comment:
could probably just be `time.Sleep(expireTTL)` (not like we have a context to cancel from)